Remotion/Data/DomainObjects/Mapping/Validation/Reflection/InheritanceHierarchyFollowsClassHierarchyValidationRule.cs

```diff
@@ -22,23 +22,23 @@ namespace Remotion.Data.DomainObjects.Mapping.Validation.Reflection
   /// <summary>
   /// Validates that the type of a class defintion is derived from base type.
   /// </summary>
-  public class InheritanceHierarchyFollowsClassHierarchyValidationRule : IClassDefinitionValidationRule
+  public class InheritanceHierarchyFollowsClassHierarchyValidationRule : ITypeDefinitionValidationRule
   {
     public InheritanceHierarchyFollowsClassHierarchyValidationRule ()
     {
 
     }
 
-    public MappingValidationResult Validate (ClassDefinition classDefinition)
+    public MappingValidationResult Validate (TypeDefinition typeDefinition)
     {
-      if (classDefinition.BaseClass !=null && !classDefinition.ClassType.IsSubclassOf(classDefinition.BaseClass.ClassType))
+      if (typeDefinition is ClassDefinition classDefinition && classDefinition.BaseClass !=null && !classDefinition.Type.IsSubclassOf(classDefinition.BaseClass.Type))
```

Comment:
needs either an interface buddy or implementation for itnerface, too. it's okay to split this into separate cases given the complexity. if so, split to have exit condtion separate separate + test